src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -9,49 +9,62 @@
 import ro.msg.learning.shop.model.Order;
 import ro.msg.learning.shop.model.OrderDetail;
 import ro.msg.learning.shop.model.Stock;
-import ro.msg.learning.shop.repository.OrderDetailsRepository;
 import ro.msg.learning.shop.repository.OrderRepository;
 import ro.msg.learning.shop.repository.ProductRepository;
 import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.strategy.LocationStrategy;
+import ro.msg.learning.shop.strategy.MostAbundantStrategy;
+import ro.msg.learning.shop.strategy.SingleLocationStrategy;
+import ro.msg.learning.shop.util.StockMapper;
 
-import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
 import java.util.Set;
-import java.util.UUID;
-import java.util.stream.Collectors;
 
 @Service
-@AllArgsConstructor
 public class OrderService {
 
     private final OrderRepository orderRepository;
-    private final ProductRepository productRepository;
     private final StockRepository stockRepository;
-    private final OrderDetailsRepository orderDetailsRepository;
-
+    private final LocationStrategy locationStrategy;
+
+    public OrderService(@Value("${strategy.type}") String strategyType, OrderRepository orderRepository, ProductRepository productRepository, StockRepository stockRepository, StockMapper stockMapper) {
+        this.orderRepository = orderRepository;
+        this.stockRepository = stockRepository;
+        switch (strategyType) {
+            case "SINGLE" -> locationStrategy = new SingleLocationStrategy(stockRepository, productRepository);
+            case "ABUNDANT" ->
+                    locationStrategy = new MostAbundantStrategy(stockRepository, productRepository, stockMapper);
+            default -> throw new NoStocksAvailableException();
```

Comment:
This will be confusing for the end user. Is the exception really a NoStockAvailableException? Or it should be something more specific, like StrategyNotImplementedException?